Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -39,9 +39,51 @@ public abstract class ClientBuilderBase<TClient>
         /// </summary>
         protected ServiceMetadata ServiceMetadata { get; }
 
+        /// <summary>
+        /// The universe domain to connect to, or null to use the default universe domain <see cref="ServiceMetadata.DefaultUniverseDomain"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <see cref="UniverseDomain"/> is used to build the endpoint to connect to, unless <see cref="Endpoint"/>
+        /// is set, in which case <see cref="Endpoint"/> will be used.
+        /// </para>
+        /// <para>
+        /// If default credentials or one of <see cref="GoogleCredential"/>, <see cref="CredentialsPath"/> or <see cref="JsonCredentials"/>
+        /// is used, <see cref="GoogleCredential.GetUniverseDomain"/> should be the same as <see cref="UniverseDomain"/> or both should be
+        /// <see cref="ServiceMetadata.DefaultUniverseDomain"/>.
+        /// </para>
+        /// </remarks>
+        public string UniverseDomain { get; set; }
+
+        /// <summary>
+        /// Effective, and known, universe domain to connect to.
```

Comment:
What's the TL;DR of when this would be null? (How should that be interpreted by the caller? That we don't know the universe domain?)